workers.go

```diff
@@ -84,41 +66,72 @@ func (bs *Bitswap) taskWorker(ctx context.Context, id int) {
 	}
 }
 
-func (bs *Bitswap) sendBlocks(ctx context.Context, env *engine.Envelope) {
-	// Blocks need to be sent synchronously to maintain proper backpressure
-	// throughout the network stack
-	defer env.Sent()
+func (bs *Bitswap) logOutgoingBlocks(env *engine.Envelope) {
+	if ce := sflog.Check(zap.DebugLevel, "sent message"); ce == nil {
+		return
+	}
 
-	msgSize := 0
-	msg := bsmsg.New(false)
+	self := bs.network.Self()
 
 	for _, blockPresence := range env.Message.BlockPresences() {
 		c := blockPresence.Cid
 		switch blockPresence.Type {
 		case pb.Message_Have:
-			log.Infof("Sending HAVE %s to %s", c.String()[2:8], env.Peer)
+			log.Debugw("sent message",
+				"type", "HAVE",
+				"cid", c,
+				"local", self,
+				"to", env.Peer,
+			)
 		case pb.Message_DontHave:
-			log.Infof("Sending DONT_HAVE %s to %s", c.String()[2:8], env.Peer)
+			log.Debugw("sent message",
+				"type", "DONT_HAVE",
+				"cid", c,
+				"local", self,
+				"to", env.Peer,
+			)
 		default:
 			panic(fmt.Sprintf("unrecognized BlockPresence type %v", blockPresence.Type))
 		}
 
-		msgSize += bsmsg.BlockPresenceSize(c)
-		msg.AddBlockPresence(c, blockPresence.Type)
 	}
 	for _, block := range env.Message.Blocks() {
-		msgSize += len(block.RawData())
-		msg.AddBlock(block)
-		log.Infof("Sending block %s to %s", block, env.Peer)
+		log.Debugw("sent message",
+			"type", "BLOCK",
+			"cid", block.Cid(),
+			"local", self,
+			"to", env.Peer,
+		)
 	}
+}
 
-	bs.sentHistogram.Observe(float64(msgSize))
-	err := bs.network.SendMessage(ctx, env.Peer, msg)
+func (bs *Bitswap) sendBlocks(ctx context.Context, env *engine.Envelope) {
+	// Blocks need to be sent synchronously to maintain proper backpressure
+	// throughout the network stack
+	defer env.Sent()
+
+	err := bs.network.SendMessage(ctx, env.Peer, env.Message)
 	if err != nil {
-		// log.Infof("sendblock error: %s", err)
-		log.Errorf("SendMessage error: %s. size: %d. block-presence length: %d", err, msg.Size(), len(env.Message.BlockPresences()))
+		log.Debugw("failed to send blocks message",
```

Comment:
Should this be logged at error level?

Reply:
Failing to send to a peer isn't an error for _us_. They may just not be reachable.

(this is actually why I started digging into this code, I saw this error in a test)